### ISSUE TITLE
PEP 702: Dedent multi-line strings used for deprecation message

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6150]
See discussion [here](https://discuss.python.org/t/pep-702-marking-deprecations-using-the-type-system/23036/53).

```python
from typing_extensions import deprecated


@deprecated(
    """
        on_event is deprecated, use lifespan event handlers instead.

        Read more about it in the
        [FastAPI docs for Lifespan Events](https://fastapi.tiangolo.com/advanced/events/).
        """
)
def on_event():
    ...
```

**Comment by erictraut** [comment]
cc @tiangolo & @JelleZijlstra

**Comment by JelleZijlstra** [comment]
See also python/peps#3482.

**Comment by erictraut** [comment]
I changed pyright to apply the same logic that it uses for docstring formatting. This mainly performs dedenting and trimming.

**Comment by tiangolo** [comment]
Nice!

**Comment by erictraut** [comment]
This is included in pyright 1.1.332, which I just published. It will also be included in a future release of pylance.
